src/users/users.service.ts

```diff
@@ -0,0 +1,84 @@
+import { Injectable, HttpException, HttpStatus, Inject } from '@nestjs/common';
+import * as bcrypt from 'bcrypt';
+import { User } from './interface/users.interface';
+import { USERS_DAO, IUsersDAO } from '../DAO/usersDAO';
+
+export const USERS_SERVICE = 'USERS SERVICE';
+export interface IUsersService {
+  findOne(firstName: string): Promise<User | null>;
+  findByEmail(email: string): Promise<User | null>;
+  findById(id: number): Promise<User | null>;
+  findAll(): Promise<User[]>;
+  saveUser(user: User): Promise<User>;
+  saveUnregisteredUser(user: User): Promise<User>;
+  updateUser(user: User): Promise<User>;
+  removeUser(id: number): Promise<User>;
+}
+
+@Injectable()
+export class UsersService implements IUsersService {
+  constructor(@Inject(USERS_DAO) private readonly usersDAO: IUsersDAO) {}
+
+  async findOne(firstName: string): Promise<User | null> {
+    return await this.usersDAO.findByFirstName(firstName);
+  }
+
+  async findByEmail(email: string): Promise<User | null> {
+    return await this.usersDAO.findOne(email);
+  }
+
+  async findById(id: number): Promise<User | null> {
+    return await this.usersDAO.findById(id);
+  }
+
+  async findAll(): Promise<User[]> {
+    const allUsers: User[] = await this.usersDAO.find();
+    allUsers.forEach((user) => {
+      user.password = undefined;
+    });
+    return allUsers;
+  }
+
+  async saveUser(user: User): Promise<User> {
+    const hashedPassword: string = await bcrypt.hash(user.password, 10);
+    user.password = hashedPassword;
+    return await this.usersDAO.save(user);
+  }
+
+  async saveUnregisteredUser(user: User): Promise<User> {
```

Comment:
We no need to have two methods to save user entity. `save` and `update` methods should be enough, I believe

Reply:
Done